File: Back/models/order.js
import mongoose from "mongoose";
import User from "./user.js";
import Product from "./product.js";
const {Schema, model} = mongoose

const orderSchema = new Schema(
    {
        user: {
            type: User,
            required: true
        },
        product: [{
            type: Product,
            required: true
        }],
        status: {
            type: String,
            required: true
        },
        responder: {
            type: User,
            required: false
        }
    },
    {
        timestamps: true
    }
);

export default model('Order', orderSchema);

File: Front/app/src/main/java/com/AndroidProject/app/appcomponents/models/Order.kt
package com.androidproject.app.appcomponents.models

import com.google.gson.annotations.SerializedName
import java.io.Serializable

data class Order(
    @SerializedName("_id")
    var id: String? = null,
    @SerializedName("user")
    var user: User? = null,
    @SerializedName("product")
    var product: ArrayList<Product>? = null,
    @SerializedName("quantity")
    var quantity: ArrayList<Int>? = null,
    @SerializedName("status")
    var status: String? = null,
    @SerializedName("price")
    var price: Float? = null,
    @SerializedName("responder")
    var responder: User?= null
):Serializable
